autometa/common/metagenome.py

```diff
@@ -537,88 +595,140 @@ def write_ranks(self, rank='superkingdom'):
-    parser.add_argument(
-        '--kmer-normalized',
-        help='Perform CLR transform on k-mer frequencies if provided. (</path/to/kmers.normalized.tsv>)',
-    )
-    parser.add_argument(
-        '--kmer-fpath',
-        help='</path/to/kmers.tsv>',
-    )
+    This handles filtering by length and taxon as well as ORF calling,
+    and various metagenome statistics.
+    """,
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('assembly', help='Path to metagenome assembly (nucleotide fasta).')
+    parser.add_argument('outdir', help='Output directory to store annotations.')
+    parser.add_argument('nucls',
+        help='Path to nucleotide ORFs corresponding to `assembly.` '
+        '(Will write to path if ORFs do not exist).',
+        type=str)
+    parser.add_argument('prots',
+        help='Path to amino acid ORFs corresponding to `assembly.` '
+        '(Will write to path if ORFs do not exist).',
+        type=str)
+    parser.add_argument('--filter',
+        help='Filter to apply to metagenome. You may supply multiple filters.',
+        choices=['length','taxon'], nargs='*')
+    parser.add_argument('--cutoff', help='Cutoff to apply to length filter',
+        default=3000,
+        type=int,
+        metavar='<int>')
+    parser.add_argument('--length-fname',
+        help='Filename to assign fasta after applying length filter.',
+        default='`assembly`.filtered.fna')
     parser.add_argument('--taxon-method', default='majority_vote',
         choices=['majority_vote'])
-    parser.add_argument('--vote-fname', help='<taxon-vote.tsv filename>',
+    parser.add_argument('--taxon-fname',
+        help='Filename to assign voted taxonomy annotation.',
         default='taxonomy_vote.tsv')
+    parser.add_argument('--ncbi',
+        help='Path to NCBI databases directory.',
+        default=NCBI_DIR)
+    parser.add_argument('--pickle', help='Whether to serialize taxonomy-specific files',
+        action='store_true', default=False)
+    parser.add_argument('--blast',
+        help='Path to diamond blast results.tsv. (outfmt=6) '
+        '(Will write to path if it does not exist).',
+        default=None)
+    parser.add_argument('--hits',
+        help='Path to diamond blast hits.pkl.gz. '
+        '(Will write to path if it does not exist).',
+        default=None)
     # Eventually will need to create a subparser for the taxon assignment methods
     # to include help information and required parameters according to method.
-    parser.add_argument('--cpus', help='num cpus to use', default=0)
-    parser.add_argument('--noparallel', help="Do not use GNU parallel",
-        action='store_false', default=True)
-    parser.add_argument('--force', help="overwrite existing files",
+    parser.add_argument('--cpus', help='Number of cpus to use when performing '
+        'annotations (Default will use all possible if `--parallel` is supplied).',
```

Comment:
Is this correct syntax? I mean the quote at the end of the line, and another quote directly after the newline. Seems like something that will cause a syntax error when run by Python.

Reply:
Since this is within the grouping `(` python will parse this as one string if a new line is encountered.

You can test this behavior with this example:


```python
# Success
line = ('here is a sentence containing multiple '
        'lines to display as one line implicitly')
print(line)

# Fails
line = 'here is a sentence containing multiple '
        'lines to display as one line implicitly'
print(line)

```

Reply:
Wow, I never knew this. Is this specific to Python 3?

Reply:
It appears to work with python2.7 as well. I'm not sure when this became possible